### PR TITLE
Add endpoints to LIP 0052

### PR DESCRIPTION
**Pull request by mjerkov**
Resolves [Issue 424](https://github.com/LiskHQ/lips/issues/424)

### CODE REVIEW

File: proposals/lip-0052.md [hunk 1]
```diff
@@ -2295,7 +2295,23 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
 
 ### Endpoints for Off-Chain Services
 
-This section specifies the non-trivial or recommended endpoints of the NFT module and does not include all endpoints. TBA
+This section specifies the non-trivial or recommended endpoints of the NFT module and does not include all endpoints. 
+
+#### hasNFT
+
+This function checks whether an NFT is owned by a specific user.
```
Comment (sergeyshemyakov) [inline proposals/lip-0052.md:2302]: I would be slightly more specific and aligned with the implementation and say:
"This function checks whether an NFT exists and is owned by a specific user."

File: proposals/lip-0052.md [hunk 2]
```diff
@@ -2295,7 +2295,23 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+
+#### getNFT
+
+This function returns NFT substore information for a provided NFT ID.
```
Comment (sergeyshemyakov) [inline proposals/lip-0052.md:2306]: The implementation also returns the locking module from the user substore. I would modify the sentence to:
"This function returns NFT and user substore information for a provided NFT ID."

File: proposals/lip-0052.md [hunk 3]
```diff
@@ -2295,7 +2295,23 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+
+#### getNFTs 
+
+This function returns NFT substore information about all NFTs owned by a specific user.
```
Comment (sergeyshemyakov) [inline proposals/lip-0052.md:2310]: Again, the locking module is included.

**Review by sergeyshemyakov — CHANGES_REQUESTED** [review]
I think it is fine not to include the pseudocode for endpoints especially since they are already implemented. However we should make sure that endpoint description aligns with the actual logic in SDK.

**Review by vjaiman — CHANGES_REQUESTED** [review]
Left a minor comment otherwise looks good.

File: proposals/lip-0052.md (hunk 1 above)
Comment (mjerkov) [reply proposals/lip-0052.md:2302]: Done.

File: proposals/lip-0052.md (hunk 2 above)
Comment (mjerkov) [reply proposals/lip-0052.md:2306]: Done.

File: proposals/lip-0052.md (hunk 3 above)
Comment (mjerkov) [reply proposals/lip-0052.md]: Done.

**Review by vjaiman — APPROVED** [review]
Looks good.

File: proposals/lip-0052.md [hunk 4]
```diff
@@ -2295,7 +2295,80 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+                "module": attribute.module,
+                "attributes": attribute.attributes
+            }
+            for attribute in NFTStore[nftID].attributesArray
+    ]
+    nftIDInfo = {
+        "owner": owner,
+        "attributesArray": attributesArray
+    }
+    if len(owner) == LENGTH_ADDRESS:
```
Comment (sergeyshemyakov) [inline proposals/lip-0052.md:2331]: I see now that this check is not in the SDK (see [here](https://github.com/LiskHQ/lisk-sdk/blob/883430a25a1f0744892fb275854f29aa1af42209/framework/src/modules/nft/endpoint.ts#L116)). But from what I can see the check is necessary for the case of escrowed NFTs. I think you should let the SDK know that the endpoint has a mistake and should be updated.

**Review by sergeyshemyakov — CHANGES_REQUESTED** [review]
One fix is still needed.

File: proposals/lip-0052.md (hunk 4 above)
Comment (mjerkov) [reply proposals/lip-0052.md:2331]: Yes, I was aware of that miss by SDK, but it seemed minor. On the other look, it is not hard to incorporate it in the SDK, so I will notify them about it.

Comment (mjerkov) [reply proposals/lip-0052.md:2331]: As pointed out by @has5aan, this issue is already resolved in SDK (see [here](https://github.com/LiskHQ/lisk-sdk/pull/8716)), so I am resolving this conversation.

**Review by vjaiman — CHANGES_REQUESTED** [review]

File: proposals/lip-0052.md [hunk 5]
```diff
@@ -2295,7 +2295,80 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+```python
+def hasNFT(address: Address, nftID: NFTID) -> bool:
+    if NFTStore[nftID] does not exist:
+        return False
+    return NFTStore[nftID].owner == address
+```
+
+#### getNFT
+
+This function checks, for a provided NFT ID, whether an NFT exists and returns NFT and user substores information.
```
[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0052.md:2313]: Yes, it is probably more readable that way, will make the change. Thx

[the remark this answers is not in the document]
Comment (mjerkov) [reply proposals/lip-0052.md]: Done, please check.

File: proposals/lip-0052.md [hunk 6]
```diff
@@ -2295,7 +2295,80 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+            "attributesArray": attributesArray,
+            "lockingModule": lockingModule
+        }
+        ownedNFTIDsInfo.append(ownedNFTIDInfo)
+    return ownedNFTIDsInfo
+```
+
+#### getEscrowedNFTIDs
+
+This function returns a list of all escrowed native NFTs for a given chain ID.
```
Comment (vjaiman) [inline proposals/lip-0052.md:2365]:
```suggestion
This function returns a list of all escrowed native NFTs for a given `chainID`.
```

File: proposals/lip-0052.md [hunk 7]
```diff
@@ -2295,7 +2295,80 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
...
+```python
+def hasNFT(address: Address, nftID: NFTID) -> bool:
+    if NFTStore[nftID] does not exist:
+        return False
+    return NFTStore[nftID].owner == address
+```
+
+#### getNFT
+
+This function returns NFT and user substores information for a provided NFT ID.
```
Comment (vjaiman) [inline proposals/lip-0052.md:2313]:
```suggestion
This function returns NFT and user substores information for a provided `nftID`.
```

**Review by vjaiman — APPROVED** [review]
Looks all good. Left a minor suggestion.

**Review by janhack — APPROVED** [review]
Editorial approval

**Pull request merged**